### PR TITLE
chore(react-jsx-runtime): simplify createElement method

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

Follow up on https://github.com/microsoft/fluentui/pull/27472#discussion_r1166478425

1. simplifies `createElement` method

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d2f17a39e1ccd8b106a442a77a4a143432cb9d54'>d2f17a39e1ccd8b106a442a77a4a143432cb9d54</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-jsx-runtime/src/createElement.ts [hunk 1]
```diff
@@ -10,40 +10,29 @@ export function createElement<P extends {}>(
   props?: P | null,
   ...children: React.ReactNode[]
 ): React.ReactElement<P> | null {
-  if (!isSlotComponent(props)) {
-    return React.createElement(type, props, ...children);
+  return isSlotComponent(props)
```
Comment (layershifter) [inline packages/react-components/react-jsx-runtime/src/createElement.ts:13]: Let's also rename it to something like `isSlotThatContainsRenderFunction` or `hasRenderFunction`.

File: packages/react-components/react-jsx-runtime/src/createElement.ts [hunk 2]
```diff
@@ -10,40 +10,29 @@ export function createElement<P extends {}>(
...
+    ? createElementFromSlot(type, props, children)
+    : React.createElement(type, props, ...children);
+}
+
+function createElementFromSlot<P extends UnknownSlotProps>(
```
Comment (layershifter) [inline packages/react-components/react-jsx-runtime/src/createElement.ts:18]:
```suggestion
function createElementFromRenderFunction<P extends UnknownSlotProps>(
```

This function does not handle slots, it handles only a specific scenario with slots. Is it more accurate? (feel to keep as it is/propose better name)

File: packages/react-components/react-jsx-runtime/src/createElement.ts (hunk 1 above)
Comment (bsunderhus) [reply packages/react-components/react-jsx-runtime/src/createElement.ts:13]: Sounds good to me, I guess I kept the name as `isSlotComponent` thinking on the future implementation

File: packages/react-components/react-jsx-runtime/src/createElement.ts (hunk 2 above)
Comment (bsunderhus) [reply packages/react-components/react-jsx-runtime/src/createElement.ts:18]: Sounds good to me, I guess I kept the name as `isSlotComponent` thinking on the future implementation

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount.html">20</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Avatar-mount_base.html">824</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Avatar-mount.html">823</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Button-mount_base.html">455</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Button-mount.html">452</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Field-mount_base.html">1510</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Field-mount.html">1554</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProvider-mount_base.html">1065</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProvider-mount.html">1077</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-mount_base.html">135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-mount.html">137</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">117</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">119</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount.html">20</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/MakeStyles-mount_base.html">1281</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/MakeStyles-mount.html">1299</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Persona-mount_base.html">2387</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/Persona-mount.html">2351</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/SpinButton-mount_base.html">1863</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27573/perf-test-react-components/SpinButton-mount.html">1869</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by layershifter — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

File: change/@fluentui-react-jsx-runtime-e321b00b-0fc8-427c-bb82-36948756d961.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-jsx-runtime-e321b00b-0fc8-427c-bb82-36948756d961.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 1 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27573&commitId=493799b26bbb3347b825c6586405d70769b0fe80&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1930 screenshots unchanged, 0 screenshots with different dimensions and 1 screenshots with visible difference.


  <details><summary><b>unknown</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.size+inactive+badge.normal.chromium.png|1| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Pull request merged**